### ISSUE TITLE
[Flake] Kokoro - TestRunGCPOnly/Google_Cloud_Build_with_Kaniko fails with docker hub timeout

### ISSUE THREAD

**Issue by ahmetb** [GoogleContainerTools/skaffold#6467]
Log: https://source.cloud.google.com/results/invocations/02cf9b74-d2f9-4934-a572-03738ae1574b/targets/skaffold%2Fpresubmit/log

It seems that we are getting dial timeouts from Docker Hub occassionally. Not sure how much we can do about this.

> error building image: error building stage: failed to get filesystem from image: Get "https://production.cloudflare.docker.com/registry-v2/docker/registry/v2/blobs/sha256/29/29291e31a76a7e560b9b7ad3cada56e8c18d50a96cca8a2573e4f4689d7aca77/data?verify=1629403571-HsgGg0mJriKCwVw4uMIWG4pvZSw%3D": read tcp 192.168.10.2:52648->104.18.125.25:443: read: connection reset by peer

Full log:

```
/integration/TestRunGCPOnly/Google_Cloud_Build_with_Kaniko
time="2021-08-19T19:15:42Z" level=info msg="Namespace: skaffoldzbs68"
time="2021-08-19T19:15:42Z" level=info msg="Running [skaffold run --namespace skaffoldzbs68 --default-repo gcr.io/k8s-skaffold] in examples/gcb-kaniko"
    helper.go:239: skaffold run: exit status 1, Generating tags...
         - skaffold-example -> gcr.io/k8s-skaffold/skaffold-example:v1.29.0-49-g819129e91
        Checking cache...
         - skaffold-example: Not found. Building
        Starting build...
        Building [skaffold-example]...
        Pushing code to gs://k8s-skaffold_cloudbuild/source/k8s-skaffold-ef3ccbf2-f50f-4637-b30f-db41f1db6665.tar.gz
        Logs are available at
        https://console.cloud.google.com/m/cloudstorage/b/k8s-skaffold_cloudbuild/o/log-94815647-5674-427e-8524-960c6e496ae4.txt
        starting build "94815647-5674-427e-8524-960c6e496ae4"

        FETCHSOURCE
        Fetching storage object: gs://k8s-skaffold_cloudbuild/source/k8s-skaffold-ef3ccbf2-f50f-4637-b30f-db41f1db6665.tar.gz#1629400544010044
        Copying gs://k8s-skaffold_cloudbuild/source/k8s-skaffold-ef3ccbf2-f50f-4637-b30f-db41f1db6665.tar.gz#1629400544010044...
        / [0 files][    0.0 B/  515.0 B]
/ [1 files][  515.0 B/  515.0 B]
        Operation completed over 1 objects/515.0 B.
        BUILD
        Pulling image: gcr.io/kaniko-project/executor:latest
        latest: Pulling from kaniko-project/executor
        Digest: sha256:6ecc43ae139ad8cfa11604b592aaedddcabff8cef469eda303f1fb5afe5e3034
        Status: Downloaded newer image for gcr.io/kaniko-project/executor:latest
        gcr.io/kaniko-project/executor:latest
        INFO[0000] GET KEYCHAIN
        INFO[0000] Resolved base name golang:1.15 to builder
        INFO[0000] Retrieving image manifest golang:1.15
        INFO[0000] Retrieving image golang:1.15 from registry index.docker.io
        INFO[0000] GET KEYCHAIN
        INFO[0000] Retrieving image manifest golang:1.15
        INFO[0000] Returning cached image manifest
        INFO[0000] Retrieving image manifest alpine:3
        INFO[0000] Retrieving image alpine:3 from registry index.docker.io
        INFO[0000] GET KEYCHAIN
        INFO[0000] Retrieving image manifest alpine:3
        INFO[0000] Returning cached image manifest
        INFO[0000] Built cross stage deps: map[0:[/app]]
        INFO[0000] Retrieving image manifest golang:1.15
        INFO[0000] Returning cached image manifest
        INFO[0000] Retrieving image manifest golang:1.15
        INFO[0000] Returning cached image manifest
        INFO[0000] Executing 0 build triggers
        INFO[0000] Checking for cached layer gcr.io/k8s-skaffold/skaffold-example/cache:67275f3247d12eec9311bc48930ee6906314d53fa1b8a1c313c7f9f070e1adfc...
        INFO[0000] GET KEYCHAIN
        INFO[0001] Using caching version of cmd: RUN go build -gcflags="${SKAFFOLD_GO_GCFLAGS}" -o /app main.go
        INFO[0001] Unpacking rootfs as cmd COPY main.go . requires it.
        INFO[0018] COPY main.go .
        INFO[0018] Taking snapshot of files...
        INFO[0018] ARG SKAFFOLD_GO_GCFLAGS
        INFO[0018] No files changed in this command, skipping snapshotting.
        INFO[0018] RUN go build -gcflags="${SKAFFOLD_GO_GCFLAGS}" -o /app main.go
        INFO[0018] Found cached layer, extracting to filesystem
        INFO[0018] Saving file app for later use
        INFO[0018] Deleting filesystem...
        INFO[0019] Retrieving image manifest alpine:3
        INFO[0019] Returning cached image manifest
        INFO[0019] Retrieving image manifest alpine:3
        INFO[0019] Returning cached image manifest
        INFO[0019] Executing 0 build triggers
        INFO[0019] Unpacking rootfs as cmd COPY --from=builder /app . requires it.
        error building image: error building stage: failed to get filesystem from image: Get "https://production.cloudflare.docker.com/registry-v2/docker/registry/v2/blobs/sha256/29/29291e31a76a7e560b9b7ad3cada56e8c18d50a96cca8a2573e4f4689d7aca77/data?verify=1629403571-HsgGg0mJriKCwVw4uMIWG4pvZSw%3D": read tcp 192.168.10.2:52648->104.18.125.25:443: read: connection reset by peer
        ERROR
        ERROR: build step 0 "gcr.io/kaniko-project/executor:latest" failed: step exited with non-zero status: 1
    panic.go:617: cloud build failed: FAILURE
    --- FAIL: TestRunGCPOnly/Google_Cloud_Build_with_Kaniko (32.58s)
/integration/TestRunGCPOnly
--- FAIL: TestRunGCPOnly (449.17s)

=== Slow Tests ===
449.17s	TestRunGCPOnly
114.73s	TestBuildKanikoInsecureRegistry
82.29s	TestRunGCPOnly/kaniko
73.11s	TestRunGCPOnly/jib_gradle_in_googlecloudbuild
67.62s	TestRunGCPOnly/jib_in_googlecloudbuild
52.86s	TestRunGCPOnly/buildpacks_on_Cloud_Build
50.62s	TestRunGCPOnly/kaniko_microservices
48.17s	TestDiagnoseRenderApply
48.17s	TestDiagnoseRenderApply/DiagnoseRenderApply
47.87s	TestDevGracefulCancel
45.29s	TestBuildInCluster/TestBuildInCluster
45.29s	TestBuildInCluster
32.64s	TestBuildGCBWithExplicitRepo
32.58s	TestRunGCPOnly/Google_Cloud_Build_with_Kaniko
27.40s	TestRunGCPOnly/kaniko_with_sub_folder
24.94s	TestRunGCPOnly/Google_Cloud_Build_with_sub_folder
21.95s	TestRenderApplyHelmDeployment/DiagnoseRenderApply
21.95s	TestRenderApplyHelmDeployment
21.89s	TestBuildKanikoWithExplicitRepo
21.40s	TestDevGracefulCancel/multiple_deployers
exit status 1
make: *** [integration-tests] Error 1
Makefile:147: recipe for target 'integration-tests' failed
Makefile:258: recipe for target 'integration-in-docker' failed
make: *** [integration-in-docker] Error 2
```

**Comment by tejal29** [comment]
shd this be fixed by #6757 @aaron-prindle ?

**Comment by aaron-prindle** [comment]
No, unfortunately that fix was specific to that test - `skaffoldyamls_test.go`.  I am not sure what the root cause is for this issue with these docker dial timeouts.
